pkg/route/validation/validation.go

```diff
@@ -342,6 +370,42 @@ func validateWildcardPolicy(host string, policy routev1.WildcardPolicyType, fldP
 	return nil
 }
 
+var (
+	notAllowedHTTPHeaderResponse    = []string{"strict-transport-security"}
+	notAllowedHTTPHeaderResponseSet = sets.NewString(notAllowedHTTPHeaderResponse...)
+)
+
+func validateHeaders(path *field.Path, headerValues []routev1.RouteHTTPHeader) *field.Error {
+	for key, value := range headerValues {
+		if value.Action.Set != nil && len(value.Action.Set.Value) != 0 && len(value.Action.Set.Value) > 4096 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, fmt.Sprintf("value exceeds the maximum length which is 2048"))
+		}
+		if value.Action.Set != nil && len(value.Action.Set.Value) == 0 {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, fmt.Sprintf("minimum length of value shall be one."))
+		}
+		fmt.Printf("path %v\n", path)
+		if value.Action.Set != nil {
+			fmt.Printf("value.Action.Set %v\n", value.Action.Set)
+		}
+		if value.Action.Set != nil && len(value.Action.Set.Value) != 0 {
+			fmt.Printf("value.Action.Set.Value %v\n", value.Action.Set.Value)
+			fmt.Printf("strings.Contains(value.Action.Set.Value, \"req.hdr\") %v\n", strings.Contains(value.Action.Set.Value, "req.hdr"))
+		}
+		if path.String() == "spec.httpHeaders.actions.response" && value.Action.Set != nil && len(value.Action.Set.Value) != 0 && strings.Contains(value.Action.Set.Value, "req.hdr") {
+			return field.Invalid(path.Index(key).Child("action.set.value"), value.Name, fmt.Sprintf("sample fetcher req.hdr is not allowed to be set in response headers."))
+		}
+		if len(value.Name) != 0 && notAllowedHTTPHeaderResponseSet.Has(strings.ToLower(value.Name)) {
+			return field.NotSupported(path.Index(key).Child("name"), value.Name, []string{fmt.Sprintf("All headers except %v", strings.Join(notAllowedHTTPHeaderResponse, ","))})
+		}
+		r := regexp.MustCompile("^(?:%(?:%|(?:\\{[-+]?[QXE](,[-+]?[QXE])*\\})?\\[(?:req\\.hdr\\([0-9A-Za-z-]+\\)|ssl_c_der)(?:,(?:lower|base64))*\\])|[^%[:cntrl:]])*$")
```

Comment:
Give `r` a more descriptive name and declare it outside the function so that we don't recompile it every time we run the validation.

Reply:
BTW, you can change `(,[-+]?[QXE])` to `(?:,[-+]?[QXE])`.

Reply:
done